Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -342,6 +342,41 @@ MainWindow::MainWindow(const QStringList &keywords, bool verbose, QWidget* paren
           this, &MainWindow::filterOperations);
   loadPlugins();
 
+  //If 3mf plugin is present, add action to save scene
+  Polyhedron_demo_io_plugin_interface* io_3mf_plugin = nullptr;
+  for(Polyhedron_demo_io_plugin_interface* plugin : io_plugins)
+  {
+    if(plugin->name() == "3mf_io_plugin")
+    {
+      io_3mf_plugin = plugin;
+      break;
+    }
+  }
+  if(io_3mf_plugin )
+  {
+    QAction* actionSaveSceneTo3mf = new QAction("Save the Scene as a 3mf File...");
+    QString dir = QString("%1/scene").arg(def_save_dir);
+    connect(actionSaveSceneTo3mf, &QAction::triggered, this,
+            [this, dir, io_3mf_plugin](){
+
+      QString filename =
+          QFileDialog::getSaveFileName(this,
+                                       tr("Save Scene to File..."),
+                                       dir,
+                                       "*.3mf");
+
+      if(filename.isEmpty())
+        return;
+      if(!filename.endsWith(".3mf"))
+      filename.append(".3mf");
+      QList<Scene_item*> all_items;
+      for(int i = 0; i< scene->numberOfEntries(); ++i)
+        all_items.push_back(scene->item(i));
+            io_3mf_plugin->save(filename, all_items);
+    });
+    ui->menuFile->insertAction(ui->actionSa_ve_Scene_as_Script, actionSaveSceneTo3mf);
+  }
```

Comment:
The plugin `io_3mf_plugin` should add itself this action into `MainWindow`, in its `init` function.

Reply:
But there is no init() function in a pure io_plugin. Doing so would require either to add an init function to all io_plugins, either to make the io_3mf_plugin inherit from the plugin_interface, only to do that. Which solution do you recommend ?

Reply:
As we break the compatibility, it is a good time to add an `init` function to IO plugins as well.
And the `Q_DECLARE_INTERFACE` of `Three/include/CGAL/Three/Polyhedron_demo_io_plugin_interface.h` should be changed, to change the version number at the end. Pick `1.90` as the version number, for example.